File: Solutions/0232- Implement Queue using Stacks/Solution.py
class MyQueue:

    def __init__(self):
        self.main_stack = []
        self.temp_stack = []

    def push(self, x: int) -> None:
        self.main_stack.append(x)

    def pop(self) -> int:
        number = self.main_stack[0]
        del self.main_stack[0]

        return number


    def peek(self) -> int:
        return self.main_stack[0]

    def empty(self) -> bool:
        return len(self.main_stack) == 0

File: Solutions/0232- Implement Queue using Stacks/Solution.java
import java.util.Stack;

class MyQueue {
    Stack<Integer> stack = new Stack<>();
    Stack<Integer> temp = new Stack<>();

    public void push(int x) {
        stack.add(x);
    }

    public int pop() {
        int size = stack.size();

        for (int i = 0; i < size - 1; i++)
            temp.add(stack.pop());

        int number = stack.pop();

        while (!temp.isEmpty())
            stack.add(temp.pop());

        return number;
    }

    public int peek() {
        int size = stack.size();

        for (int i = 0; i < size - 1; i++)
            temp.add(stack.pop());

        int number = stack.peek();
        temp.add(stack.pop());

        while (!temp.isEmpty())
            stack.add(temp.pop());

        return number;
    }

    public boolean empty() {
        return stack.isEmpty();
    }
}